File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
let OPEN = ["(", "[", "{"]

function solution(s) {
    let answer = 0;
    let str = s.concat(s);
    for(let i = 0; i < s.length; i++){
        const stack = [];
        let isCorrect = true;
        for(let j = i; j < i + s.length; j++){
            if(OPEN.includes(str[j])){
                stack.push(str[j]);
            } 
            else{
                if(stack.length === 0){
                    isCorrect = false;
                    break;
                }
                let top = stack[stack.length - 1];
                if(top === "(" && str[j] === ")"){
                    stack.pop()
                }
                else if(top === "{" && str[j] === "}"){
                    stack.pop()
                }
                else if(top === "[" && str[j] === "]"){
                    stack.pop()
                }
                else{
                    isCorrect = false;
                    break;
                }
            }
            
        }
        if(stack.length !== 0){
                isCorrect = false;
            }
        if(isCorrect) answer++;
    }

    return answer;
}   

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def solution(s):
    answer = 0
    for i in range(len(s)):
        stack = []
        for j in range(len(s)):
            item = s[(i+j)%len(s)]
            if item == '[' or item == '(' or item == '{':
                stack.append(item)
            else:
                if len(stack) == 0:
                    stack.append(item)
                    break
                    
                if item == ']' and stack[-1] == '[':
                    stack.pop()
                elif item == ')' and stack[-1] == '(':
                    stack.pop()
                elif item == '}' and stack[-1] == '{':
                    stack.pop()
        
        if len(stack) == 0:
            answer += 1
    
    return answer
                    